umake/frameworks/ide.py

```diff
@@ -755,3 +755,84 @@ def post_install(self):
                         exec=os.path.join(self.install_path, "Code"),
                         comment=_("Visual Studio focused on modern web and cloud"),
                         categories="Development;IDE;"))
+
+
+class SpringToolsSuite(umake.frameworks.baseinstaller.BaseInstaller):
+    def __init__(self, category):
+        super().__init__(name="Spring Tools Suite",
+                         description=_("Spring Tools Suite IDE"),
+                         download_page="https://spring.io/tools/sts/all",
+                         dir_to_decompress_in_tarball='sts-bundle/sts-*',
+                         desktop_filename='STS.desktop',
+                         category=category, only_on_archs=['i386', 'amd64'],
+                         packages_requirements=['openjdk-7-jdk'],
+                         icon_filename='icon.xpm')
+
+    def download_provider_page(self):
+        logger.debug("Download application provider page")
+        DownloadCenter([DownloadItem(self.download_page)], self.get_metadata, download=False)
+
+    @MainLoop.in_mainloop_thread
+    def get_metadata(self, result):
+        """Download files to download"""
+        logger.debug("Parse download metadata")
+
+        error_msg = result[self.download_page].error
+        if error_msg:
+            logger.error("An error occurred while downloading {}: {}".format(self.download_page, error_msg))
+            UI.return_main_screen(status_code=1)
+
+        in_download = False
+        url_found = False
+        line_content = result[self.download_page].buffer.getvalue().decode()
+        (url_found, in_download) = self.parse_download_link(line_content, in_download)
+
+        if not url_found:
+            logger.error("Download page changed its syntax or is not parsable")
+            UI.return_main_screen(status_code=1)
+
+    def parse_download_link(self, line, in_download):
+        """Parse STS download link"""
+        url_found = False
+        in_download = False
+
+        if '>Linux<' in line and '64BIT' in line:
+            in_download = True
+        if in_download:
+            regexp = r'GTK, 64BIT.+?href="(.*?)"'
+            if get_current_arch() == "i386":
+                regexp = r'GTK, 32BIT.+?href="(.*?)"'
+            r = re.compile(regexp, re.DOTALL)
```

Comment:
you can use re.search(regexp, line) instead of having compile and search.
also, I wouldn't parse multiple lines. The links are already different, and it's better to keep this the same as the other classes.
It should be enough to search for:
"linux-gtk{}.tar.gz".format(arch) in line
with arch begin replaced with "" if 32bit, "-x86_64" if 64bit (see the nodejs implementation here: https://github.com/ubuntu/ubuntu-make/blob/master/umake/frameworks/nodejs.py#L52)

then the regex would simply look for something like:
p = re.search(r'href="(.*.tar.gz)"', line)